adr/2025-09-05-app-requirements-validation.md

```diff
@@ -0,0 +1,97 @@
+---
+title: App requirements validation
+date: 2025-09-05
+area: core
+tags: [app-system, validation, requirements]
+---
+
+## Context
+
+The Shopware app system allows third-party applications to extend platform functionality. However, certain apps may require specific environment conditions or infrastructure capabilities to function properly. Without proper validation of these requirements, apps might fail at runtime or behave unpredictably, leading to poor user experience and difficult troubleshooting.
+
+Examples of such requirements include:
+- Public accessibility for webhook endpoints
+- Secure HTTPS connections for data exchange
+- Network connectivity for external service integrations
+- Specific server configurations or capabilities
+
+Currently, the app system lacks a standardized way to define and validate these requirements before app installation or updates, which can lead to:
+- Silent failures in production environments
+- Difficult debugging when apps don't work as expected
+- Poor developer experience when requirements are not clearly communicated
+- Security vulnerabilities when apps are installed in inappropriate environments
+
+## Decision
+
+We will implement a best-effort app requirements validation system that allows apps to declare their requirements in the manifest and validates these requirements during installation and update processes.
+
+### Key Components
+
+**1. Manifest Requirements Declaration**
+Apps can declare requirements using a new `<requirements>` element in their manifest XML with boolean attributes:
+
+```xml
+<requirements requires-public-access="true" />
```

Comment:
How will this look with multiple rules?

```suggestion
<requirements requires-public-access="true" requires-specific-other-rule="true" requires-specific-second-other-rule="true" />
```

vs

```suggestion
<requirements>
  <requires>public-access</requires>
  <requires>specific-other-rule</requires>
  <requires>specific-second-other-rule</requires>    
</requirements>  
```

Do we need the "true" ?

Reply:
So it is verbosity vs. readability. It was just a thought I had while reading the code. You can also go ahead with the stuff you already have. 😄 👍🏼